### PR TITLE
Add document for ml command.

### PR DESCRIPTION
**Pull request by jngz-es**
Signed-off-by: Jing Zhang <jngz@amazon.com>

### Description
Add document for the generic ml command in PPL.
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/849
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
Please update `docs/user/ppl/index.rst`

**Comment by jngz-es** [comment]
Thanks, will update it.

**Comment by penghuo** [comment]
add deprecation on Kmeans and AD docs?

**Review by penghuo — APPROVED** [review]
Thanks.

File: docs/user/ppl/cmd/ml.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+* number_of_trees(integer): optional. Number of trees in the forest. The default value is 30.
+* shingle_size(integer): optional. A shingle is a consecutive sequence of the most recent records. The default value is 8.
+* sample_size(integer): optional. The sample size used by stream samplers in this forest. The default value is 256.
+* output_after(integer): optional. The number of points required by stream samplers before results are returned. The default value is 32.
+* time_decay(double): optional. The decay factor used by stream samplers in this forest. The default value is 0.0001.
+* anomaly_rate(double): optional. The anomaly rate. The default value is 0.005.
+* time_field(string): mandatory. It specifies the time field for RCF to use as time-series data.
+* date_format(string): optional. It's used for formatting time_field field. The default formatting is "yyyy-MM-dd HH:mm:ss".
+* time_zone(string): optional. It's used for setting time zone for time_field filed. The default time zone is UTC.
+* category_field(string): optional. It specifies the category field used to group inputs. Each category will be independently predicted.
```
[the remark this answers is not in the document]
Comment (joshuali925) [reply docs/user/ppl/cmd/ml.rst:36]: it's a temporary solution implemented in PPL https://github.com/opensearch-project/sql/pull/952

**Review by penghuo — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]

**Pull request merged**